### PR TITLE
feat(docs): qmake plugin reference

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by cmatsuoka — APPROVED** [review]

File: docs/common/craft-parts/reference/plugins/qmake_plugin.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+**Type:** int
+**Default:** 5
+
+Sets the Qt major version. The default is Qt 5, set to 6 for Qt 6 projects.
+
+
+Environment variables
+---------------------
+
+The plugin sets the QT_SELECT environment variable to ref:`qt-major-version`.
```
Comment (tigarmo) [inline docs/common/craft-parts/reference/plugins/qmake_plugin.rst:45]: Do you mean `qmake-major-version`?

File: docs/common/craft-parts/reference/plugins/qmake_plugin.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+dependencies the source requires to build are to be provided.
+
+How it works
+------------
+
+During the build step the plugin performs the following actions:
+
+* Run ``qmake`` in the build directory to setup the ``Makefiles``, the
+  project is configured with any ``qmake-parameters`` that might have
+  been set. If ``qmake-project-file`` has been set, ``qmake`` refer to
```
Comment (tigarmo) [inline docs/common/craft-parts/reference/plugins/qmake_plugin.rst:65]:
```suggestion
  been set. If ``qmake-project-file`` has been set, ``qmake`` refers to
```

File: docs/common/craft-parts/reference/plugins/qmake_plugin.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+* ``make`` calls the ``install`` target with ``DESTDIR`` set to
+  ``$CRAFT_PART_INSTALL``.
+
+Examples
+--------
+
+The following snippet declares a part using the ``qmake`` plugin for a
+local source that contains a ``.pro`` project file:
+
+.. code-block:: yaml
```
Comment (tigarmo) [inline docs/common/craft-parts/reference/plugins/qmake_plugin.rst:77]: Ideally this example would also showcase some of the options

**Review by tigarmo — CHANGES_REQUESTED** [review]

File: docs/common/craft-parts/reference/plugins/qmake_plugin.rst (hunk 3 above)
Comment (sergiusens) [reply docs/common/craft-parts/reference/plugins/qmake_plugin.rst:77]: ok, I was trying to find a practical example but qmake has been deprecated, qmake 6 was the last qmake and everyone migrated to cmake

Comment (tigarmo) [reply docs/common/craft-parts/reference/plugins/qmake_plugin.rst:77]: I think even just setting `qmake-major-version: 6` should be ok

**Review by tigarmo — APPROVED** [review]

**Pull request merged**